src/main/kotlin/Main.kt

```diff
@@ -0,0 +1,29 @@
+fun main(args: Array<String>) {
+    scenario(12,8)
+}
+
+fun scenario(numberOfCars : Int, capacity : Int) {
+    /*
+    * Creates a given a number of cars, every other car wants to drive
+    * A road network is created with the given capacity
+    * The road network is analyzed and it is printed whether a car is delayed or not
+    */
+    val listOfCars : MutableList<Car> = mutableListOf()
+    val road = Network(capacity,listOfCars)
+    for (i in 1..numberOfCars) {
+        if (i % 2 == 0) {
+            Car(i,true, listOfCars)
```

Comment:
I really hope you don't think I'm trying to stress you out because I'm not trying that. But I still think this is more complicated that it has to be. What about this:

```kotlin
val listOfCars : MutableList<Car> = mutableListOf()
for (i in 1..numberOfCars) {
    val newCar = Car(i, i % 2 == 0)
    listOfCars.add(newCar)
}
val road = Network(capacity,listOfCars)
```